core/src/main/scala/com/lightbend/paradox/template/PageTemplateSystem.scala

```diff
@@ -20,20 +20,33 @@ import java.io.File
 import java.util.{ Map => JMap }
 import org.stringtemplate.v4.misc.STMessage
 import org.stringtemplate.v4.{ STErrorListener, STRawGroupDir, ST }
+import collection.concurrent.TrieMap
+
+object CachedTemplates {
+  def apply(dir: File, templateName: String): PageTemplate = {
+    cache.get(templateName) match {
+      case Some(t) => t
+      case _ =>
+        val newTemplate = new PageTemplate(dir, templateName)
+        cache(templateName) = newTemplate
+        newTemplate
+    }
+  }
+
+  val cache: TrieMap[String, PageTemplate] = TrieMap()
+}
 
 /**
  * Page template writer.
  */
-class PageTemplate(directory: File, startDelimiter: Char = '$', stopDelimiter: Char = '$', name: String = "page") {
+class PageTemplate(directory: File, name: String = PageTemplate.DefaultName, startDelimiter: Char = '$', stopDelimiter: Char = '$') {
```

Comment:
Here's what I used for testing https://github.com/eed3si9n/tech-hub-landing-page/tree/wip/layout_check, which didn't work for me.